src/main/scala/com/ovotech/micrometer/Reporter.scala

```diff
@@ -62,22 +65,48 @@ object Reporter {
       sem <- Semaphore[F](1)
     } yield new ReporterImpl[F](mx, metricPrefix, globalTags, mutable.Map.empty, sem)
 
+  private class GaugeKey(private val name: String, tags: Tags) {
```

Comment:
Why not using a case class here?

Reply:
I typically don't use a case class when I override `equals` and `hashCode`. Structural equality won't work here because `Tags.equal` is sensitive to ordering, which seems inappropriate because we allow constructions of tags via passing a `Map`, which is unordered.